tst/EnergyPlus/unit/CurveManager.unit.cc

```diff
@@ -998,3 +998,156 @@ TEST_F( EnergyPlusFixture, Tables_OneIndependentVariable_Linear_EvaluateCurveTyp
 
 		EXPECT_FALSE( has_err_output() );
 }
+
+TEST_F( EnergyPlusFixture, TableLookupObject_ExcessArguments_WarningTest ) {
+
+	int NumOfTables = 5;
+
+	PerfCurve.allocate( NumOfTables );
+	TableLookup.allocate( NumOfTables );
+
+	// Table 1 One ind variable
+	PerfCurve( 1 ).TableIndex = 1;
+	PerfCurve( 1 ).Var1Max = 20.0;
+	PerfCurve( 1 ).Var1Min = 0.0;
+	PerfCurve( 1 ).Var2Max = 0.0;
+	PerfCurve( 1 ).Var2Min = 0.0;
+	PerfCurve( 1 ).CurveType = Linear;
+	TableLookup( PerfCurve( 1 ).TableIndex ).NumIndependentVars = 1;
+	PerfCurve( 1 ).ObjectType = CurveType_TableOneIV;
+	PerfCurve( 1 ).Name = "Table 1";
+	TableLookup( 1 ).NumX1Vars = 2;
+	TableLookup( 1 ).InterpolationOrder = 2;
+	TableLookup( 1 ).NumX1Vars = 2;
+	TableLookup( 1 ).X1Var.allocate( TableLookup( 1 ).NumX1Vars );
+	TableLookup( 1 ).X1Var( 1 ) = 0.0;
+	TableLookup( 1 ).X1Var( 2 ) = 1.0;
+	TableLookup( 1 ).TableLookupZData.allocate( 1, 1, 1, 1, 2 );
+	TableLookup( 1 ).TableLookupZData( 1, 1, 1, 1, 1 ) = 0.0;
+	TableLookup( 1 ).TableLookupZData( 1, 1, 1, 1, 2 ) = 1.0;
+
+	TableLookupObject( 1, 1.0, 0.0 );
+
+	std::string const error_string = delimited_string( {
+		"   ** Warning ** TableLookupObject: Table:OneIndependentVariable\"Table 1\"",
+		"   **   ~~~   ** ...Excess number of independent variables (2) passed to subroutine when only 1 is required. The excess arguments are ignored.",
+	} );
+
+	EXPECT_TRUE( compare_err_stream( error_string, true ) );
```

Comment:
:+1: :+1:

Reply:
Will this warning happen once, or every time the curve is called?

Reply:
@lgu1234 increments the warning counter here: https://github.com/NREL/EnergyPlus/pull/6429/files#diff-6bab88f88cad01231c7611b162099cdaR4648

And blocks for it above, here: https://github.com/NREL/EnergyPlus/pull/6429/files#diff-6bab88f88cad01231c7611b162099cdaR4645

So it will happen a max of once per curve index, as far as I can tell.

Reply:
@mjwitte It only happens once for each curve.